testing/scenario_app/bin/utils/adb_logcat_filtering.dart

```diff
@@ -40,14 +46,15 @@ library;
 /// with lazy parsing.
 extension type const AdbLogLine._(Match _match) {
   // RegEx that parses into the following groups:
-  // 1. Everything up to the severity (I, W, E, etc.).
-  //    In other words, any whitespace, numbers, hyphens, colons, and periods.
-  // 2. The severity (a single uppercase letter).
-  // 3. The name of the process (up to the colon).
-  // 4. The message (after the colon).
+  // 1. The time of the log message, such as `02-22 13:54:39.839`.
+  // 2. The process ID.
+  // 3. The thread ID.
+  // 4. The character representing the severity of the log message, such as `I`.
+  // 5. The tag, such as `ActivityManager`.
+  // 6. The actual log message.
   //
   // This regex is simple versus being more precise. Feel free to improve it.
-  static final RegExp _pattern = RegExp(r'([^A-Z]*)([A-Z])\s([^:]*)\:\s(.*)');
+  static final RegExp _pattern = RegExp(r'(\d+-\d+\s[\d|:]+\.\d+)\s+(\d+)\s+(\d+)\s(\w)\s(\S+)\s*:\s*(.*)');
```

Comment:
Can you put those comments inline in the regex?  I'm not sure if the dart regex parser supports them.

Reply:
I cannot unfortunately (Dart regex support lacking).